src/hooks/useAuthForm.tsx

```diff
@@ -0,0 +1,44 @@
+import { useForm } from 'react-hook-form';
+import { FieldValues, SubmitHandler } from 'react-hook-form';
+interface ISignIn {
+  email: string;
+  password: string;
+}
+interface ISignUp extends ISignIn {
+  firstname: string;
+  lastname: string;
+  passwordConfirmation: string;
+}
+
+export interface CommonFieldValues extends FieldValues {
```

Comment:
Always Use "I" as a first letter for interfaces. 
It shouldn't works in such way, we don't have passwordConfirmation or firstname fields in signIn model.
putting it into tech debt